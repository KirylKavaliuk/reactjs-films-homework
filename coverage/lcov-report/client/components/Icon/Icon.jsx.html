<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/Icon/Icon.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">client/components/Icon</a> Icon.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">28x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">28x</span>
<span class="cline-any cline-yes">27x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
&nbsp;
import withConditionalRendering from 'utils/rendering';
&nbsp;
import Star from 'icons/star.svg';
import Search from 'icons/search.svg';
import Play from 'icons/play.svg';
import Cancel from 'icons/cancel.svg';
import Circle from 'icons/circle.svg';
import Arrow from 'icons/arrow.svg';
import gridView from 'icons/grid-view.svg';
import listView from 'icons/list-view.svg';
&nbsp;
const icons = {
  star: Star,
  search: Search,
  play: Play,
  cancel: Cancel,
  circle: Circle,
  arrow: Arrow,
  'grid-view': gridView,
  'list-view': listView,
};
&nbsp;
const Icon = ({ name, className }) =&gt; {
  const Component = icons[name.toLowerCase()];
&nbsp;
  if (Component) {
    return &lt;Component className={ className }/&gt;;
  }
&nbsp;
  return &lt;Circle/&gt;;
};
&nbsp;
Icon.defaultProps = {
  className: '',
};
&nbsp;
Icon.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
};
&nbsp;
export default withConditionalRendering(Icon, 'name');
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 01 2019 11:31:09 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
