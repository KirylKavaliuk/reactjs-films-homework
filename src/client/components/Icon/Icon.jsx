import React from 'react';
import PropTypes from 'prop-types';
import withConditionalRendering from 'utils/withConditionalRendering';

import Star from 'icons/star.svg';
import Search from 'icons/search.svg';
import Play from 'icons/play.svg';
import Cancel from 'icons/cancel.svg';
import Circle from 'icons/circle.svg';
import Arrow from 'icons/arrow.svg';
import gridView from 'icons/grid-view.svg';
import listView from 'icons/list-view.svg';

const icons = {
  star: Star,
  search: Search,
  play: Play,
  cancel: Cancel,
  circle: Circle,
  arrow: Arrow,
  'grid-view': gridView,
  'list-view': listView,
};

const Icon = ({ name, className }) => {
  const Component = icons[name.toLowerCase()];

  if (Component) {
    return <Component className={ className }/>;
  }

  return <Circle/>;
};

Icon.defaultProps = {
  className: '',
  name: 'star',
};

Icon.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
};

export default withConditionalRendering(Icon, ['name']);
