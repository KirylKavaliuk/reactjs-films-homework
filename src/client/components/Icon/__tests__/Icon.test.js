import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../Icon';

it('<Icon/>', () => {
  const tree = renderer.create(<Icon name='play'/>).toJSON();

  expect(tree).not.toBeNull();
  expect(tree).toMatchSnapshot();
});

it('<Icon/>', () => {
  const tree = renderer.create(
    <Icon name='gun' className='Component__class-name__hash'/>,
  ).toJSON();

  expect(tree).not.toBeNull();
  expect(tree).toMatchSnapshot();
});

it('<Icon/>', () => {
  const tree = renderer.create(<Icon/>).toJSON();

  expect(tree).toBeNull();
  expect(tree).toMatchSnapshot();
});
