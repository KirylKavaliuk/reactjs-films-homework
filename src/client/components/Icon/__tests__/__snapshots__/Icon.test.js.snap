// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon/> 1`] = `
<svg
  className=""
  viewBox="0 0 58 58"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="29"
    cy="29"
    r="29"
  />
  <path
    d="M44 29L22 44V14z"
    fill="#fff"
  />
  <path
    d="M22 45a.999.999 0 0 1-1-1V14a.999.999 0 0 1 1.564-.826l22 15a1.001 1.001 0 0 1-.001 1.652l-22 15A1.002 1.002 0 0 1 22 45zm1-29.107v26.215L42.225 29 23 15.893z"
    fill="#fff"
  />
</svg>
`;

exports[`<Icon/> 2`] = `
<svg
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 0C114.837 0 0 114.837 0 256s114.837 256 256 256 256-114.837 256-256S397.163 0 256 0z"
  />
</svg>
`;

exports[`<Icon/> 3`] = `null`;
